### PR TITLE
Update to latest otel-java snapshot

### PR DESCRIPTION
**Pull request by iNikem**

### CODE REVIEW

File: instrumentation/opentelemetry-api-beta/src/main/java/io/opentelemetry/instrumentation/auto/opentelemetryapi/metrics/ApplicationDoubleCounter.java [hunk 1]
```diff
@@ -37,6 +37,11 @@ public void add(double delta, Labels labels) {
     agentDoubleCounter.add(delta, LabelBridging.toAgent(labels));
   }
 
+  @Override
+  public void add(double v) {
```
Comment (anuraaga) [inline instrumentation/opentelemetry-api-beta/src/main/java/io/opentelemetry/instrumentation/auto/opentelemetryapi/metrics/ApplicationDoubleCounter.java:41]: Don't know how the coverage of these tests is but if it's possible to add that'd be nice. When adding these bridges I needed them to be (mostly) sure my bridging is sound. Though admittedly these seem simpler than a lot of the tracing / context related stuff.

**Review by anuraaga — APPROVED** [review]
Welcome to tower bridge, we have a great view of the Thames from here. Let's keep these engines running :)

File: instrumentation/opentelemetry-api-beta/src/main/java/io/opentelemetry/instrumentation/auto/opentelemetryapi/metrics/ApplicationDoubleCounter.java (hunk 1 above)
Comment (iNikem) [reply instrumentation/opentelemetry-api-beta/src/main/java/io/opentelemetry/instrumentation/auto/opentelemetryapi/metrics/ApplicationDoubleCounter.java:41]: I am not aware of any metric-related tests in our repo. Including these bridges

Comment (trask) [reply instrumentation/opentelemetry-api-beta/src/main/java/io/opentelemetry/instrumentation/auto/opentelemetryapi/metrics/ApplicationDoubleCounter.java:41]: See https://github.com/open-telemetry/opentelemetry-java-instrumentation/blob/master/instrumentation/opentelemetry-api-beta/src/test/groovy/MeterTest.groovy

Comment (iNikem) [reply instrumentation/opentelemetry-api-beta/src/main/java/io/opentelemetry/instrumentation/auto/opentelemetryapi/metrics/ApplicationDoubleCounter.java:41]: I was unable to understand in 2 minutes how to better do that. If this is blocker, I will have to think about it tomorrow, with fresh head

**Comment by iNikem** [comment]
I had to add a bunch of dependencies on context-propagation artifact to force gradle to take the required time-based snapshot. Without it, some unknown version was used locally.

File: instrumentation/opentelemetry-api-beta/src/main/java/io/opentelemetry/instrumentation/auto/opentelemetryapi/metrics/ApplicationDoubleCounter.java (hunk 1 above)
Comment (trask) [reply instrumentation/opentelemetry-api-beta/src/main/java/io/opentelemetry/instrumentation/auto/opentelemetryapi/metrics/ApplicationDoubleCounter.java:41]: No problem to open issue(s) for anything unresolved in this PR, better to merge it sooner to resolve #1102

Comment (iNikem) [reply instrumentation/opentelemetry-api-beta/src/main/java/io/opentelemetry/instrumentation/auto/opentelemetryapi/metrics/ApplicationDoubleCounter.java:41]: Then feel free to approve and merge if build is green :)

Comment (trask) [reply instrumentation/opentelemetry-api-beta/src/main/java/io/opentelemetry/instrumentation/auto/opentelemetryapi/metrics/ApplicationDoubleCounter.java:41]: I created #1105 for this

**Pull request merged**